catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DurableFileSystemFileConsumer.java

```diff
@@ -18,12 +18,11 @@
 import org.apache.camel.component.file.GenericFileEndpoint;
 import org.apache.camel.component.file.GenericFileOperations;
 import org.apache.camel.component.file.GenericFileProcessStrategy;
-import org.apache.commons.io.monitor.FileAlterationObserver;
 
 public class DurableFileSystemFileConsumer extends AbstractDurableFileConsumer {
   private DurableFileAlterationListener listener;
 
-  private FileAlterationObserver observer;
+  private AsyncFileAlterationObserver observer;
```

Comment:
❗️Changing the `observer` type makes it impossible to reuse the persisted state from a previous DDF install, which could be a big problem. We should find a way to keep the persisted observer compatible and document how to carry the state over to the new DDF install when upgrading.

Reply:
I looked into this some more. While there was an initialization method within the Observer it was never actually called. If we started CDM (in-place) it would act as it does with this implementation where the observer would start with an "empty state". I'm not sure how this would work for upgrades but when I was testing with the vanilla 2.13.x version it seemed that removing and re-adding the same monitored directory within the CDM configurations, ddf would load the serialized state from the file system and not reingest the files into the catalog.

I reimplemented the init method and a implemented a destroy method in case someone would want to initialize / reset the internal state, but like the old implementation they aren't currently called (aside from testing).